### ISSUE TITLE
SplitControl With Heatmap Layers

### ISSUE THREAD

**Issue by maboualidev** [jupyter-widgets/ipyleaflet#532]
The following code will not produce proper results:

```
mymap = Map(center=(lat, lon), zoom=4)
right_layer = Heatmap(locations=right_locations, max_zoom=max_zoom, radius=radius, max=max_value)
left_layer = Heatmap(locations=left_locations, max_zoom=max_zoom, radius=radius, max=max_value)
split_map_control = SplitMapControl(left_layer=left_layer , right_layer=right_layer )
mymap.add_control(split_map_control )
mymap
```

Both layers are shown on one side (right_side).

**Comment by martinRenou** [comment]
Thanks for reporting! The splitcontrol might need some improvements. I'll try to find time to look into it.

**Comment by maboualidev** [comment]
Thank for the great tool. Is there another way to do it?

Also if I have a 2D numpy array, is there a way to convert it to a tile layer and show it on leaflet without saving it to GeoTiff and creating tiles?

**Comment by davidbrochart** [comment]
> Also if I have a 2D numpy array, is there a way to convert it to a tile layer and show it on leaflet without saving it to GeoTiff and creating tiles?

You can have a look at https://github.com/jupyter-widgets/ipyleaflet/blob/master/examples/Numpy.ipynb.

**Comment by martinomiani** [comment]
> The following code will not produce proper results:
> 
> ```
> mymap = Map(center=(lat, lon), zoom=4)
> right_layer = Heatmap(locations=right_locations, max_zoom=max_zoom, radius=radius, max=max_value)
> left_layer = Heatmap(locations=left_locations, max_zoom=max_zoom, radius=radius, max=max_value)
> split_map_control = SplitMapControl(left_layer=left_layer , right_layer=right_layer )
> mymap.add_control(split_map_control )
> mymap
> ```
> 
> Both layers are shown on one side (right_side).

The problem exists also when using other layers in combination with SplitControl, such as GeoJSON, GeoData and Choropleths

**Comment by perone** [comment]
I'm also having the same issue with `SplitMapControl`, it seems to be very bugged with a lot of different layers, with GeoJSON object it shows only right side as well.

**Comment by martinRenou** [comment]
Yes, unfortunately, the Spitmap control does not work properly. This is actually an issue with the SplitMapControl plugin for LeafletJS that we use https://github.com/QuantStack/leaflet-splitmap#readme. We improved it a bit recently. It needs a bit more testing. Anyone with JavaScript knowledge is very welcome to fix this!

**Comment by msameeruddin** [comment]
Any update on this?

**Comment by giswqs** [comment]
Is it possible to display a widget (e.g., legend) on top of the SplitControl?

![image](https://user-images.githubusercontent.com/5016453/135155985-af376317-0df9-415c-98ec-6bfaec627084.png)

**Comment by data-henrik** [comment]
Any update? It seems that the plugin behind has not much of activity.

**Comment by martinRenou** [comment]
Yes, unfortunately it doesn't have much activity. If someone wants to pick that up I would be happy to review a PR.

**Comment by pietervdwijk** [comment]
I have no experience with javascript, but could this line in the plugin be the culprit of the issue that left-side does not show the correct layers?
https://github.com/QuantStack/leaflet-splitmap/blob/2fa719b418c16d0b25d098d7a8400afe4354872b/src/leaflet-splitmap.js#L150

![Image](https://github.com/user-attachments/assets/3ae33bdc-2e5c-4fcf-a74c-809c90b73bb9)
